### ISSUE TITLE
Some alternate capitalizations of site URLs trigger the invalid URL page

### ISSUE THREAD

**Issue by OceanBagel** [TASVideos/tasvideos#726]
If the first letter of any URL subdirectory doesn't match the case of the correct URL, the invalid URL page is triggered. However, the rest of the URL is case-insensitive.

Example:
[https://tasvideos.org/EmulatorResources/LibTAS](https://tasvideos.org/EmulatorResources/LibTAS) is the correct URL.
[https://tasvideos.org/EmulatorResources/libTAS](https://tasvideos.org/EmulatorResources/libTAS) lands on the [invalid URL page](https://tasvideos.org/Wiki/PageNotFound?possibleUrl=EmulatorResources%2FLibTAS). (Expected: lands on the same page as above.)
[https://tasvideos.org/EmulatorResources/Libtas](https://tasvideos.org/EmulatorResources/Libtas) lands on the correct page.

**Comment by bigbass1997** [comment]
Further, at least with the libtas example, when you attempt to edit any of its subpages, such as https://tasvideos.org/EmulatorResources/LibTAS/Usage, you're redirected to a blank screen (https://tasvideos.org/Wiki/Edit?path=EmulatorResources%2FlibTAS%2FUsage, changing the url to an uppercase LibTAS gets it to work though).

**Comment by vadosnaprimer** [comment]
Don't remember if it was ever asked but do we want to make wiki page _links_ case insensitive? Would probably solve a lot of problems.

**Comment by adelikat** [comment]
The bug here is that scenario 3 (https://tasvideos.org/EmulatorResources/Libtas) is NOT behaving like scenario 2.

We don't want urls to be case insensitive for SEO reasons.  And we don't want to simply redirect for you because if there is a link out in the wild, I want them to fix it, so we don't have endless redirects impacting the server.  The page you land on has the correct link, so you can fix it.  The site itself will always link to the correct page.

**Comment by meshuggahtas** [comment]
> The bug here is that scenario 3 (https://tasvideos.org/EmulatorResources/Libtas) is NOT behaving like scenario 2.

Are you sure? An average visitor probably thinks that the page which titled **libTAS** would be reachable by scenario 2 **libTAS**

**Comment by adelikat** [comment]
Yes, I'm sure, because if they think that and type it into the url, it should suggest the correct page which they can then click

**Comment by meshuggahtas** [comment]
Oh I understand now. This happens with all pages, that is https://tasvideos.org/Emulatorresources/Libtas and other variations for scenario 3 issue.

**Comment by Masterjun3** [comment]
I don't understand this discussion. What do we even want the correct page to be?
Our site code says all pages need to start with an uppcase letter (with unrelated exceptions).
But the database (and thus our pages and links) all show the page as `libTAS`.

What is the intent?

**Comment by adelikat** [comment]
I can answer the intent. The intent is that libTAS is wrong and needs to be fixed. Which it has.

**Comment by adelikat** [comment]
What I'm unhappy about now is that invalid URL patterns get the redirect page.  But case misspellings do not.  So /BizHawk and /Bizhawk go to the page, but /bizhawk does not.

I'm not sure what I want to do about this yet.  Kicking the can to somewhere down the road.  At least wiki pages will not be created that do not match the intended naming conventions with the current set up

**Comment by Masterjun3** [comment]
Whatever we do with this, currently the DB page name for the BizHawk page is `/Bizhawk`, and many referrers link to the correct `/Bizhawk` and many link to the wrong `/BizHawk`. So if we make the wrong casing to redirect to the PageNotFound page, we will need to change a lot of wiki pages.
